rclpy/rclpy/node.py

```diff
@@ -1194,6 +1197,10 @@ def add_on_set_parameters_callback(
         :param callback: The function that is called whenever parameters are being validated
                          for the node.
         """
+        if not callable(callback):
+            raise TypeError('Callback must be callable, got {}', type(callback))
+        if not isinstance(callback([Parameter('dummy', value=0)]), SetParametersResult):
```

Comment:
i think this is not good solution, we need to check the signature without calling the callback. (this bothers the application.)